lib/kube/proxy/forwarder.go

```diff
@@ -698,6 +716,31 @@ func (f *Forwarder) setupContext(authCtx auth.Context, req *http.Request, isRemo
 	}, nil
 }
 
+// fillDefaultKubePrincipalDetails fills the default details in order to keep
+// the correct behavior when forwarding the request to the Kubernetes API.
+// By default, if no kubernetes_users are set (which will be a majority), a
+// user will impersonate himself, which is the backwards-compatible behavior.
+// We also append teleport.KubeSystemAuthenticated to kubernetes_groups, which is
+// a builtin group that allows any user to access common API methods,
+// e.g. discovery methods required for initial client usage, without it,
+// restricted user's kubectl clients will not work.
```

Comment:
I think it's alright to ditch either the function comment or the code comments below, as they say the same thing.